### PR TITLE
C2 amr a raouf c2 s05 ro fb invite

### PR DESCRIPTION
**Pull request by AmrARaouf**
[#356]

@Omar-Hossam @karimngr @Mirna-Yacout @SalmaFG

### CODE REVIEW

File: arability/app/models/authentication.rb [hunk 1]
```diff
@@ -81,4 +81,51 @@ def self.get_common_twitter_followers(current_gamer)
...
+  #   Returns the access token of a certain provider for a specified gamer
+  # Params
+  #   id: id of the gamer
+  #   provider: name of the provider
+  # Success:
+  #   Returns the provider's access token for the gamer
+  # Failure:
+  #   None
+  def self.get_token(id, provider)
+    auth = Authentication.find_by_gamer_id_and_provider(id,provider)
```
Comment (omar-mohyeldin) [inline arability/app/models/authentication.rb]: leave space after comma

File: arability/app/views/games/game.html.erb [hunk 2]
```diff
@@ -170,14 +170,12 @@ function setButtons(){
...
-     <% end %>
+      $('#zone').append('<a style="margin: 14px; padding:4px;" class="btn btn-success pull-left" onclick="backToGame()"><%= t(:back_to_game) %>');
+      $('#zone').append('<div class="provider_buttons" style="text-align:center; display:inline-block;"><a href="/games/post_facebook"><img src="/assets/facebook_share_ar.png" style="height:150px;padding:10px;"></a><a href="http://www.twitter.com/share?text=http://localhost:3000/ تعالى إلعب معي على via @SmartSoft2013"><img src="/assets/twitter_share_ar.png" style="height:150px;"></a></div>')
+    <% else I18n.locale == :en %>
+      $('#zone').append('<a style="margin: 14px; padding:4px;" class="btn btn-success pull-left" onclick="backToGame()"><%= t(:back_to_game) %>');
+      $('#zone').append('<div class="provider_buttons" style="text-align:center; display:inline-block;"><a href="/games/post_facebook"><img src="/assets/facebook_share.png" style="height:150px;padding:10px;"></a><a href="http://www.twitter.com/share?text=http://localhost:3000/ تعالى إلعب معي على via @SmartSoft2013"><img src="/assets/twitter_share.png" style="height:150px;"></a></div>')
```
Comment (omar-mohyeldin) [inline arability/app/views/games/game.html.erb]: use double quotations for the previous 4 lines

File: arability/app/views/games/game.html.erb [hunk 3]
```diff
@@ -270,47 +268,44 @@ function setButtons(){
...
 
   <% if I18n.locale == :ar %>
   <div class="zone" id="zone" style="text-align:right; direction: rtl;">
-
-    <% else %>
-    <div class="zone" id="zone">
-      <% end %> 
+  <% else %>
+  <div class="zone" id="zone">
+  <% end %>
```
Comment (omar-mohyeldin) [inline arability/app/views/games/game.html.erb:285]: adjust indentation with "if"

File: arability/app/controllers/authentications_controller.rb [hunk 4]
```diff
@@ -119,7 +119,8 @@ def facebook_callback
           redirect_to "/gamers/edit",
           flash: {success: t(:logged_in_to_fb)}
         else
-          redirect_to root_url, flash: {notice: t(:already_connected_fb)}
+          Authentication.update_token(current_gamer.id,provider,token)
```
Comment (omar-mohyeldin) [inline arability/app/controllers/authentications_controller.rb]: leave space after comma

File: arability/app/models/authentication.rb [hunk 5]
```diff
@@ -81,4 +81,51 @@ def self.get_common_twitter_followers(current_gamer)
...
+  # Params:
+  #   id: id of the gamer
+  #   provider: provider that wants to update the token
+  #   new_token: the new access token
+  # Success:
+  #   acess token for provider is changed for a specified gamer
+  # Failure:
+  #   none
+  def self.update_token(id, provider, new_token)
+    auth = Authentication.find_by_gamer_id_and_provider(id,provider)
```
Comment (omar-mohyeldin) [inline arability/app/models/authentication.rb]: leave space after comma

**Comment by omar-mohyeldin** [comment]
@AmrARaouf please modify your code

**Comment by omar-mohyeldin** [comment]
@AmrARaouf code verified

**Comment by ghost** [comment]
verified

**Comment by salmafg** [comment]
Testing verified.

File: arability/app/models/authentication.rb [hunk 6]
```diff
@@ -81,4 +81,51 @@ def self.get_common_twitter_followers(current_gamer)
...
   end
 
+  # Author
+  #   Amr Abdelraouf
+  # Description:
+  #   Returns whether or not a gamer's account is connected to facebook
+  # Params:
+  #   id: the id of the gamer
+  # Success:
+  #   Boolean is returned to ndicate whether the gamer is connected
```
Comment (salmafg) [inline arability/app/models/authentication.rb]: indicate*

File: arability/app/models/authentication.rb [hunk 7]
```diff
@@ -81,4 +81,51 @@ def self.get_common_twitter_followers(current_gamer)
...
+  # Author:
+  #   Amr Abdelraouf
+  # Description:
+  #   Updates access token (mostly when it expires)
+  # Params:
+  #   id: id of the gamer
+  #   provider: provider that wants to update the token
+  #   new_token: the new access token
+  # Success:
+  #   acess token for provider is changed for a specified gamer
```
Comment (salmafg) [inline arability/app/models/authentication.rb]: Access*

**Comment by Mirna-Yacout** [comment]
verified

**Comment by AmrARaouf** [comment]
@SalmaFG please re-review

**Comment by salmafg** [comment]
Documentation verified.

**Pull request merged**